app/client/rest/tracking.test.ts

```diff
@@ -244,13 +244,12 @@ describe('ClientTracking', () => {
 
         try {
             await client.doFetchWithTracking('https://example.com/api', options);
-            fail('Expected error to be thrown');
         } catch (error: unknown) {
             const clientError = error as ClientError;
 
-            expect((clientError.details as {message: string}).message).toBe('Custom error message');
-            expect((clientError.details as {server_error_id: string}).server_error_id).toBe('error_id_123');
-            expect((clientError.details as {status_code: string}).status_code).toBe(400);
+            expect((clientError as {message: string}).message).toBe('Custom error message');
+            expect((clientError as {server_error_id: string}).server_error_id).toBe('error_id_123');
+            expect((clientError as {status_code: number}).status_code).toBe(400);
```

Comment:
I didn't catch it the first time but you could do:

```
            expect(error).toBeInstanceOf(ClientError);

            const clientError = error as ClientError;
            expect(clientError.message).toBe('Custom error message');
            expect(clientError.server_error_id).toBe('error_id_123');
            expect(clientError.status_code).toBe(400);
```

but you'd have to import

```
import ClientError from './error';
```

instead of

```
import type ClientError from '.error';
```